tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/FilterSet.scala

```diff
@@ -53,8 +53,11 @@ object RuleWithId {
         .id(Rule.Id.of(ruleWithId.id.value))
         .name(ruleWithId.name.value)
         .conditionGroup(ruleWithId.conditionGroup match {
-          case Some(conditionGroup) => Rule.ConditionGroup.of(Rule.ConditionCombiner.valueOf(conditionGroup.conditionCombiner.toString),
-            conditionGroup.conditions.map(convertScalaConditionToJavaCondition(_)).asJava)
+          case Some(conditionGroup) => ruleWithId.condition match {
+            case Some (_) => throw new RuntimeException("Rule with id " + ruleWithId.id.value + " contains both conditionGroup and condition field")
```

Comment:
Well in scala exception should be carried explicitly by the type system and not thrown.

Can we refactor this piece of code to use `Either` ?